File: PSI-2020-backend/PSI-2020-backend/Models/Database/EducationProgram.cs
using PSI_2020_backend.Models.Database.Bases;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Reflection.Metadata.Ecma335;
using System.Threading.Tasks;

namespace PSI_2020_backend.Models.Database
{
    public class EducationProgram : ModelBase
    {
        public int Year { get; set; }
        public bool IsWIP { get; set; }
        public Degree Degree { get; set; }
        public Term Term { get; set; }
        public Field Field { get; set; }
        public Specialization Specialization { get; set; }
        public List<Course> Courses { get; set; }
        public bool IsHistorical { get; set; }
        public DateTime ModificationDate { get; set; }
        public string ModifiedBy { get; set; }
        public ModificationType ModificationType { get; set; }
    }
}

File: frontend/src/app/models/education-program.ts
import { Course } from "./course";
import { Degree } from "./degree.enum";
import { Field } from "./field";
import { Specialization } from "./specialization";
import { SubjectCard } from "./subject-card";
import { Term } from "./term.enum";

export class EducationProgram {
    id: string;
    year: number;
    isWIP: boolean;
    degree: Degree;
    term: Term;
    field: Field;
    specialization?: Specialization;
    courses: Course[];

    constructor(init?: Partial<EducationProgram>) {
        Object.assign(this, init);
    }

    static create() {
       return new EducationProgram({
           isWIP: true,
           courses: [],
       });
    }
}
